odoo/addons/base/models/ir_model.py

```diff
@@ -937,6 +914,165 @@ def _add_manual_fields(self, model):
                     model._add_field(name, field)
 
 
+class IrModelSelection(models.Model):
+    _name = 'ir.model.fields.selection'
+    _order = 'sequence, id'
+    _description = "Fields Selection"
+
+    field_id = fields.Many2one("ir.model.fields",
+        required=True, ondelete="cascade",
+        domain=[('ttype', 'in', ['selection', 'reference'])])
+    value = fields.Char(required=True)
+    name = fields.Char(translate=True, required=True)
+    sequence = fields.Integer(default=10)
+
+    _sql_constraints = [
+        ('selection_field_uniq', 'unique(field_id, value)',
+         'Selections values must be unique per field'),
+    ]
+
+    @tools.ormcache()
+    def _get_field_data(self, field_id):
+        """ Return the given field's selection. """
+        self._cr.execute("""
+            SELECT value, name
+            FROM ir_model_fields_selection
+            WHERE field_id=%s
+            ORDER BY sequence, id
+        """, (field_id,))
+        return self._cr.fetchall()
+
+    @tools.ormcache('model_name')
+    def _existing_field_data(self, model_name):
+        """ Return the given model's existing field data. """
+        self._cr.execute("""
+            SELECT f.name as field_name, s.value, s.name, s.sequence
+            FROM ir_model_fields f
+            JOIN ir_model_fields_selection s
+            ON f.id = s.field_id
+            WHERE f.model=%s
+            ORDER BY s.sequence, s.id
+            """, [model_name])
+        res = defaultdict(list)
+        for row in self._cr.dictfetchall():
+            res[row['field_name']].append(row)
+        return res
```

Comment:
The returned value is the selection data per field, so I suggest that the data contains the values of `ir.model.selection`'s fields: `field_id`, `value`, `name`, `sequence`. This will make it easier to compare it with the output of `_reflect_field_params`.

Reply:
By the way, the fact that `field_id` is currently *not* in the output makes `old_vals != new_vals` in `_reflect_model` always true :sob: